drivers/sensor/bmm150/bmm150.c

```diff
@@ -0,0 +1,606 @@
+		if(bmm150_samp_freq_table[i].reg_val == odr_val) {
+			data->odr = bmm150_samp_freq_table[i].freq;
+			return 0;
+		}
+	}
+
+	return -ENOTSUP;
+}
+#endif
+
+#if defined(CONFIG_BMM150_SAMPLING_REP_XY)
+static int bmm150_write_rep_xy(struct device *dev, int val)
+{
+	struct bmm150_data *data = dev->driver_data;
+	const struct bmm150_config *config = dev->config->config_info;
+
+	if(i2c_reg_update_byte(data->i2c, config->i2c_slave_addr,
+				BMM150_REG_REP_XY,
+				BMM150_REG_REP_DATAMASK,
+				BMM150_REPXY_TO_REGVAL(val)) < 0) {
+		return -EIO;
+	}
+
+	data->rep_xy = val;
+
+	return 0;
+}
+#endif
+
+#if defined(CONFIG_BMM150_SAMPLING_REP_Z)
+static int bmm150_write_rep_z(struct device *dev, int val)
+{
+	struct bmm150_data *data = dev->driver_data;
+	const struct bmm150_config *config = dev->config->config_info;
+
+	if(i2c_reg_update_byte(data->i2c, config->i2c_slave_addr,
+				BMM150_REG_REP_Z,
+				BMM150_REG_REP_DATAMASK,
+				BMM150_REPZ_TO_REGVAL(val)) < 0) {
+		return -EIO;
+	}
+
+	data->rep_z = val;
+
+	return 0;
+}
+#endif
+/* Reference Datasheet 4.3.2 */
+static s32_t bmm150_compensate_xy(struct bmm150_trim_regs *tregs,
+				    int16_t xy, uint16_t rhall, bool is_x)
+{
+	s8_t txy1, txy2;
+	s16_t val;
+	
+	if(xy == BMM150_XY_OVERFLOW_VAL) {
+		return INT32_MIN;
+	}
+
+	if (!rhall) {
+		rhall = tregs->xyz1;
+	}
+
+	if(is_x) {
+		txy1 = tregs->x1;
+		txy2 = tregs->x2;
+	} else {
+		txy1 = tregs->y1;
+		txy2 = tregs->y2;
+	}
+
+	val = ((s16_t)(((u16_t)((((s32_t)tregs->xyz1) << 14) / rhall)) -
+              ((u16_t)0x4000)));
+        val = ((s16_t)((((s32_t)xy) * ((((((((s32_t)tregs->xy2) *
+              ((((s32_t)val) * ((s32_t)val)) >> 7)) + (((s32_t)val) *
+              ((s32_t)(((s16_t)tregs->xy1) << 7)))) >> 9) +
+              ((s32_t)0x100000)) * ((s32_t)(((s16_t)txy2) +
+              ((s16_t)0xA0)))) >> 12)) >> 13)) + (((s16_t)txy1) << 3);
+
+	return (int32_t)val;
+
+}
+
+static s32_t bmm150_compensate_z(struct bmm150_trim_regs *tregs,
+				   s16_t z, u16_t rhall)
+{
+	s32_t val;
+
+	if (z == BMM150_Z_OVERFLOW_VAL) {
+		return INT32_MIN;
+	}
+
+	val = (((((s32_t)(z - tregs->z4)) << 15) - ((((s32_t)tregs->z3) *
+              ((s32_t)(((s16_t)rhall) - ((s16_t)tregs->xyz1)))) >> 2)) /
+              (tregs->z2 + ((s16_t)(((((s32_t)tregs->z1) *
+              ((((s16_t)rhall) << 1))) + (1 << 15)) >> 16))));	
+
+	return val;
+}
+
+static int bmm150_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+
+ 	struct bmm150_data *drv_data = dev->driver_data;
+	const struct bmm150_config *config = dev->config->config_info;
+	u16_t values[BMM150_AXIS_XYZR_MAX];
+	s16_t raw_x, raw_y, raw_z;
+	u16_t rhall;
+
+ 	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_MAGN_ANY);
+
+	if(i2c_burst_read(drv_data->i2c, config->i2c_slave_addr,
+				BMM150_REG_X_L, (u8_t *)values,
+				sizeof(values)) < 0) {
+		SYS_LOG_DBG("failed to read sample");
+		return -EIO;
+	}
+ 
+	raw_x = (s16_t)sys_le16_to_cpu(values[BMM150_AXIS_X]) >>
```

Comment:
I am using raw_x as argument in bmm150_compensate_xy(). That's why raw_x cannot be directly assigned to drv_data->sample_x